tictactoe.py

```diff
@@ -43,75 +43,126 @@
+
+
+def displayBoard():
+    print("\n", gameBoard[0:3], "     ", [1, 2, 3],
+          "\n", gameBoard[3:6], "     ", [4, 5, 6],
+          "\n", gameBoard[6:10], "     ", [7, 8, 9],
+          "\n")
+
+
+def checkPlay(userMove, userGamePiece):
+    if userMove == ("1"):
+        gameBoard[0] = userGamePiece
+    if userMove == ("2"):
+        gameBoard[1] = userGamePiece
+    if userMove == ("3"):
+        gameBoard[2] = userGamePiece
+    if userMove == ("4"):
+        gameBoard[3] = userGamePiece
+    if userMove == ("5"):
+        gameBoard[4] = userGamePiece
+    if userMove == ("6"):
+        gameBoard[5] = userGamePiece
+    if userMove == ("7"):
+        gameBoard[6] = userGamePiece
+    if userMove == ("8"):
+        gameBoard[7] = userGamePiece
+    if userMove == ("9"):
+        gameBoard[8] = userGamePiece
+
+
+def checkWin(gameBoard):
+    if gameBoard[0] == gameBoard[1] == gameBoard[2] and gameBoard[0] != DEFAULTSPACE:
+        print(WINMESSAGE)
+        return True
+    if gameBoard[3] == gameBoard[4] == gameBoard[5] and gameBoard[3] != DEFAULTSPACE:
+        print(WINMESSAGE)
+        return True
+    if gameBoard[6] == gameBoard[7] == gameBoard[8] and gameBoard[6] != DEFAULTSPACE:
+        print(WINMESSAGE)
+        return True
+    if gameBoard[0] == gameBoard[3] == gameBoard[6] and gameBoard[0] != DEFAULTSPACE:
+        print(WINMESSAGE)
+        return True
+    if gameBoard[1] == gameBoard[4] == gameBoard[7] and gameBoard[1] != DEFAULTSPACE:
+        print(WINMESSAGE)
+        return True
+    if gameBoard[2] == gameBoard[5] == gameBoard[8] and gameBoard[2] != DEFAULTSPACE:
+        print(WINMESSAGE)
+        return True
+    if gameBoard[0] == gameBoard[4] == gameBoard[8] and gameBoard[0] != DEFAULTSPACE:
+        print(WINMESSAGE)
+        return True
+    if gameBoard[2] == gameBoard[4] == gameBoard[6] and gameBoard[2] != DEFAULTSPACE:
+        print(WINMESSAGE)
+        return True
+    else:
+        spaces = 0
+        for space in gameBoard:
+            if space != DEFAULTSPACE:
+                spaces += 1
+        if spaces == 9:
+            print(DRAWMESSAGE)
+            return True
+        return False
+
+
+def aiMove():
+    if gameBoard[4] == DEFAULTSPACE:
+        gameBoard[4] = OGAMEPIECE
+    elif gameBoard[0] == DEFAULTSPACE:
+        gameBoard[0] = OGAMEPIECE
+    elif gameBoard[2] == DEFAULTSPACE:
+        gameBoard[2] = OGAMEPIECE
+    elif gameBoard[6] == DEFAULTSPACE:
+        gameBoard[6] = OGAMEPIECE
+    elif gameBoard[8] == DEFAULTSPACE:
+        gameBoard[8] = OGAMEPIECE
+    elif gameBoard[1] == DEFAULTSPACE:
+        gameBoard[1] = OGAMEPIECE
+    elif gameBoard[3] == DEFAULTSPACE:
+        gameBoard[3] = OGAMEPIECE
+    elif gameBoard[5] == DEFAULTSPACE:
+        gameBoard[5] = OGAMEPIECE
+    elif gameBoard[7] == DEFAULTSPACE:
+        gameBoard[7] = OGAMEPIECE
+
+
+def game1():
     playerWin = False
+    while playerWin is False:
+        userX = input("User X. Enter position number: ")
```

Comment:
I found a bug where you if enter anything other than a position number it skips your turn and allows the AI to play by itself, you should add a check that only lets a player enter a number or it says "invalid entry".

For example if I entered the letter A instead of a position it would skip my turn and let the AI keep playing